test/fpm_test/test_filesystem.f90

```diff
@@ -96,11 +98,102 @@ subroutine check_string(error, actual, expected)
 
         if (actual /= expected) then
             call test_failed(error, &
-                "Character value missmatch "//&
+                "Character value mismatch "//&
                 "expected '"//expected//"' but got '"//actual//"'")
         end if
 
     end subroutine check_string
 
 
+    subroutine test_mkdir_rmdir(error)
+
+        !> Error handling
+        type(error_t), allocatable, intent(out) :: error
+
+        logical :: is_win
+
+        is_win = (get_os_type() == OS_WINDOWS)
+
+        if (is_win) then
+            call check_mkdir(error, "tmpdir\subdir")
+          else
+            call check_mkdir(error, "tmpdir/subdir")
+        end if
```

Comment:
We could use [`join_path`](https://fortran-lang.github.io/fpm/proc/join_path.html) to simplify these checks
```suggestion

        call check_mkdir(error, join_path("tmpdir","subdir"))
```